### PR TITLE
Oracle setter functions

### PR DESCRIPTION
**Pull request by haythem96**
This PR add the code to set the following parameters:
- Batch oracle address
- Oracle locking period
- Oracle dispute period

Should be reviewed/merged after merging #54 .

### CODE REVIEW

File: contracts/Oracle.sol [hunk 1]
```diff
@@ -28,6 +28,13 @@ contract Oracle is Ownable {
     /// @dev mapping between batch and it price at specific timestmap. A batch is the hash of underlying, collateral, strike and expiry.
     mapping(bytes32 => mapping(uint256 => Price)) internal batchPriceAt;
 
+    /// @notice emits an event when an oracle added for a specific batch
+    event BatchOracleAdded(bytes32 indexed batch, address oracle);
+    /// @notice emits an event when a locking period added for a specific oracle
+    event OracleLockingPeriodAdded(address indexed oracle, uint256 lockingPeriod);
+    /// @notice emits an event when a dispute period added for a specific oracle
+    event OracleDisputePeriodAdded(address indexed oracle, uint256 disputePeriod);
```
Comment (antoncoding) [inline contracts/Oracle.sol:36]: Maybe `BatchOracleUpdated ` and `OracleLockingPeriodUpdated `?

File: package.json [hunk 2]
```diff
@@ -83,5 +83,7 @@
     "url": "https://github.com/opynfinance/GammaProtocol/issues"
   },
   "homepage": "https://github.com/opynfinance/GammaProtocol#readme",
-  "dependencies": {}
+  "dependencies": {
+    "ethers": "^5.0.7"
+  }
```
Comment (antoncoding) [inline package.json:88]: Maybe be in devDependencies like any other packages?

File: test/oracle.test.ts [hunk 3]
```diff
@@ -15,9 +16,57 @@ contract('Oracle', ([owner]) => {
...
+    const lockingPeriod = new BigNumber(60 * 15) // 15min
+
+    it('should revert setting oracle locking period from non-owner address', async () => {
+      await expectRevert(
+        oracle.setLockingPeriod(batchOracle, lockingPeriod, {from: random}),
+        'Ownable: caller is not the owner',
+      )
+    })
+
+    it('should set oracle locking period', async () => {
```
Comment (aparnakr) [inline test/oracle.test.ts:41]: Is there a max time locking period that can be set? We might need some restrictions. Also is the owner manually setting locking period?!

File: test/oracle.test.ts [hunk 4]
```diff
@@ -15,9 +16,57 @@ contract('Oracle', ([owner]) => {
...
+      await expectRevert(
+        oracle.setDisputePeriod(batchOracle, disputePeriod, {from: random}),
+        'Ownable: caller is not the owner',
+      )
+    })
+
+    it('should set oracle locking period', async () => {
+      await oracle.setDisputePeriod(batchOracle, disputePeriod, {from: owner})
+
+      assert.equal(
```
Comment (aparnakr) [inline test/oracle.test.ts:65]: Again, probably good to have some upper and lower bounds on dispute period. 

Are the dispute and locking periods set once or set per oracle? I’m a little confused by that.

File: test/oracle.test.ts (hunk 3 above)
Comment (haythem96) [reply test/oracle.test.ts:41]: I don't think we discussed a max time locking period ? if so what is the limit that I should test ?
and re owner, yeah the owner manually setting locking period.

File: test/oracle.test.ts (hunk 4 above)
Comment (haythem96) [reply test/oracle.test.ts:65]: yeah locking and dispute period are set per oracle in this case.

Comment (aparnakr) [reply test/oracle.test.ts:65]: maybe 3 days max, 15 minutes min? (by oracle you mean per strike-expiry-underliying-collateral? )

File: test/oracle.test.ts (hunk 3 above)
Comment (aparnakr) [reply test/oracle.test.ts:41]: maybe 3 days max, 15 minutes min?

Comment (aparnakr) [reply test/oracle.test.ts:41]: hmm so how is the locking / dispute period set before the prices are set or how does the overall flow work?

File: contracts/Oracle.sol (hunk 1 above)
Comment (haythem96) [reply contracts/Oracle.sol]: cool updated this.

File: package.json (hunk 2 above)
Comment (haythem96) [reply package.json:88]: this is removed in other PR.

File: test/oracle.test.ts (hunk 4 above)
Comment (haythem96) [reply test/oracle.test.ts:65]: a locking and dispute period is set by oracle address.

**Comment by haythem96** [comment]
Going to merge this one and add limit for periods in another PR.

**Pull request merged**